### PR TITLE
docker: add docker file

### PR DESCRIPTION
**Pull request by gibiw**
docker: add docker file
--
- add docker file for building docker image with qasectl
- add `docker` command to Makefile

---

ci: add docker build job
--
Add `build-docker-n-publish` job to the pipeline. It creates and publishs docker image to ghcr.io

### CODE REVIEW

File: .github/workflows/build-binaries.yml [hunk 1]
```diff
@@ -82,3 +82,29 @@ jobs:
...
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push Docker image
+        id: push
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: ./build/Dockerfile
+          push: true
+          tags: ghcr.io/${{ github.repository }}/qase:${{ env.TAG }}
```
Comment (NickVolynkin) [inline .github/workflows/build-binaries.yml:108]: I believe, org name goes first: `ghcr.io/qase-tms/qase:tag`

File: .github/workflows/build-binaries.yml [hunk 2]
```diff
@@ -82,3 +82,30 @@ jobs:
...
+    needs: build-n-publish
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
```
Comment (NickVolynkin) [inline .github/workflows/build-binaries.yml:98]: who is `github.actor`? One who pushed the tag?

File: .github/workflows/build-binaries.yml (hunk 1 above)
Comment (gibiw) [reply .github/workflows/build-binaries.yml]: fixed

File: .github/workflows/build-binaries.yml [hunk 3]
```diff
@@ -82,3 +82,36 @@ jobs:
...
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push Docker image
+        id: push
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: ./build/Dockerfile
+          push: true
+          tags: ghcr.io/qase-tms/qase:${{ env.TAG }}
```
Comment (NickVolynkin) [inline .github/workflows/build-binaries.yml:114]:
```suggestion
          tags: ghcr.io/qase-tms/qase:${{ env.TAG }},ghcr.io/qase-tms/qase:latest
```

**Review by NickVolynkin — APPROVED** [review]

**Pull request merged**